samples/directx/d3d10_opencl.cpp

```diff
@@ -0,0 +1,397 @@
+    }
+
+
+    void print_info(ID3D10Texture2D* pSurface, int mode, float fps, cv::String oclDevName)
+    {
+        HRESULT r;
+
+        UINT subResource = ::D3D10CalcSubresource(0, 0, 1);
+
+        D3D10_MAPPED_TEXTURE2D mappedTex;
+        r = pSurface->Map(subResource, D3D10_MAP_WRITE_DISCARD, 0, &mappedTex);
+        if (FAILED(r))
+        {
+            return;
+        }
+
+        cv::Mat m(m_height, m_width, CV_8UC4, mappedTex.pData, (int)mappedTex.RowPitch);
+
+        cv::String strMode = cv::format("%s", m_modeStr[mode].c_str());
+        cv::String strFPS = cv::format("%2.1f", fps);
+        cv::String strDevName = cv::format("%s", oclDevName.c_str());
+
+        cv::putText(m, strMode, cv::Point(0, 16), 1, 0.8, cv::Scalar(0, 0, 0));
+        cv::putText(m, strFPS, cv::Point(0, 32), 1, 0.8, cv::Scalar(0, 0, 0));
+        cv::putText(m, strDevName, cv::Point(0, 48), 1, 0.8, cv::Scalar(0, 0, 0));
+
+        m_pSurface->Unmap(subResource);
+
+        return;
+    }
+
+
+    int render()
+    {
+        try
+        {
+            if (m_shutdown)
+                return 0;
+
+            HRESULT r;
+            ID3D10Texture2D* pSurface;
+
+            r = get_surface(&pSurface);
+            if (FAILED(r))
+            {
+                return -1;
+            }
+
+            switch (m_mode)
+            {
+                case 0:
+                    // no processing
+                    break;
+
+                case 1:
+                {
+                    // process video frame on CPU
+                    UINT subResource = ::D3D10CalcSubresource(0, 0, 1);
+
+                    D3D10_MAPPED_TEXTURE2D mappedTex;
+                    r = m_pSurface->Map(subResource, D3D10_MAP_WRITE_DISCARD, 0, &mappedTex);
+                    if (FAILED(r))
+                    {
+                        return r;
+                    }
+
+                    cv::Mat m(m_height, m_width, CV_8UC4, mappedTex.pData, (int)mappedTex.RowPitch);
+
+                    if (!m_disableProcessing)
+                    {
+                        // blur D3D10 surface with OpenCV on CPU
+                        cv::blur(m, m, cv::Size(15, 15), cv::Point(-7, -7));
+                    }
+
+                    m_pSurface->Unmap(subResource);
+
+                    break;
+                }
+
+                case 2:
+                {
+                    // process video frame on GPU
+                    cv::UMat u;
+
+                    cv::directx::convertFromD3D10Texture2D(pSurface, u);
+
+                    if (!m_disableProcessing)
+                    {
+                        // blur D3D9 surface with OpenCV on GPU with OpenCL
+                        cv::blur(u, u, cv::Size(15, 15), cv::Point(-7, -7));
+                    }
+
+                    cv::directx::convertToD3D10Texture2D(u, pSurface);
+
+                    break;
+                }
+
+            } // switch
+
+            print_info(pSurface, m_mode, getFps(), m_oclDevName);
```

Comment:
What is `getFPS()` measure? I believe we need to measure processing time (or not?).